pkg/materialize/cluster.go

```diff
@@ -124,65 +126,88 @@ func (b *ClusterBuilder) Scheduling(v []interface{}) *ClusterBuilder {
+		p = append(p, i)
+	}
 
-		if b.replicationFactor != nil {
-			i := fmt.Sprintf(` REPLICATION FACTOR %v`, *b.replicationFactor)
-			p = append(p, i)
+	if len(b.availabilityZones) > 0 {
+		var az []string
+		for _, z := range b.availabilityZones {
+			f := QuoteString(z)
+			az = append(az, f)
 		}
+		a := fmt.Sprintf(`AVAILABILITY ZONES = [%s]`, strings.Join(az[:], ","))
+		p = append(p, a)
+	}
 
-		if len(b.availabilityZones) > 0 {
-			var az []string
-			for _, z := range b.availabilityZones {
-				f := QuoteString(z)
-				az = append(az, f)
-			}
-			a := fmt.Sprintf(` AVAILABILITY ZONES = [%s]`, strings.Join(az[:], ","))
-			p = append(p, a)
-		}
+	if b.introspectionInterval != "" {
+		i := fmt.Sprintf(`INTROSPECTION INTERVAL = %s`, QuoteString(b.introspectionInterval))
+		p = append(p, i)
+	}
 
-		if b.introspectionInterval != "" {
-			i := fmt.Sprintf(` INTROSPECTION INTERVAL = %s`, QuoteString(b.introspectionInterval))
-			p = append(p, i)
-		}
+	if b.introspectionDebugging {
+		p = append(p, `INTROSPECTION DEBUGGING = TRUE`)
+	}
 
-		if b.introspectionDebugging {
-			p = append(p, ` INTROSPECTION DEBUGGING = TRUE`)
-		}
+	if b.schedulingConfig.IsSet {
+		// We skip setting this in alter cluster because it
+		// will be handled independently
+		p = append(p, b.GenSchedulingConfigSql(b.schedulingConfig))
+	}
 
-		if b.schedulingConfig.OnRefresh.Enabled {
-			scheduleStatement := " SCHEDULE = ON REFRESH"
-			if b.schedulingConfig.OnRefresh.HydrationTimeEstimate != "" {
-				scheduleStatement += fmt.Sprintf(" (HYDRATION TIME ESTIMATE = %s)", QuoteString(b.schedulingConfig.OnRefresh.HydrationTimeEstimate))
-			} else if b.schedulingConfig.OnRefresh.RehydrationTimeEstimate != "" {
-				scheduleStatement += fmt.Sprintf(" (HYDRATION TIME ESTIMATE = %s)", QuoteString(b.schedulingConfig.OnRefresh.RehydrationTimeEstimate))
-			}
-			p = append(p, scheduleStatement)
-		}
+	if len(p) > 0 {
+		return strings.Join(p[:], ", ")
+	}
+	return ""
+}
 
-		if len(p) > 0 {
-			p := strings.Join(p[:], ",")
-			q.WriteString(fmt.Sprintf(`,%s`, p))
-		}
+func (b *ClusterBuilder) Create() error {
+	q := strings.Builder{}
+
+	q.WriteString(fmt.Sprintf(`CREATE CLUSTER %s`, b.QualifiedName()))
+	// Size is the only required option for managed clusters. A "REPLICAS ()" option indicates
+	// and is required for unmanaged clusters. No other options can be provided for managed
+	// clusters.
+	if b.size != "" {
+		q.WriteString(fmt.Sprintf(` (%s)`, b.GenerateClusterOptions()))
 	} else {
-		q.WriteString(` REPLICAS ()`)
+		q.WriteString(` (REPLICAS ())`)
 	}
 
 	q.WriteString(`;`)
+	return b.ddl.exec(q.String())
+}
+
+func (b *ClusterBuilder) AlterClusterScheduling(s SchedulingConfig) error {
+	q := strings.Builder{}
+	q.WriteString(fmt.Sprintf(`ALTER CLUSTER %s`, b.QualifiedName()))
+	q.WriteString(fmt.Sprintf(` SET (%s)`, b.GenSchedulingConfigSql(s)))
+	q.WriteString(`;`)
+	return b.ddl.exec(q.String())
+}
```

Comment:
@bobbyiliev the plan now is that we do AlterClusterScheduling separate from AlterCluster and always first. 

This will allow us to from a cluster with scheduling and no replication factor to a cluster without scheduling and a replication factor, and vice versa.

Note, we only really set the scheduling option on the ClusterBuilder directly in the create phase where everything can be done in a single statement.